WordPress/gutenberg
# Link to homeUrl from site editor view menu.

## What?

Replaces the siteURL link in the full site editor view menu with the homeUrl.

## Why?


siteUrl is a reference to the WordPress install, ie the folder containing wp-cron.php, wp-mail.php, etc. homeUrl is the home page of the website. On sites with WordPress installed in a sub-directory, the two values can differ.

## How?


I've added `@wordpress/core-data` as a dependency of the site editor header and used that to locate the homeUrl. I suspect it could be added equally well the the existing settings but I was unsure how to go about that.

Fixes #45474.

## Testing Instructions
1. Create a [sub-directory install](https://wordpress.org/support/article/giving-wordpress-its-own-directory/) of WordPress 
2. Visit the site editor
3. From the view dropdown, click view site

This should open the site's home page rather than the sub-directory containing the install/

## Screenshots or screencast
